lib/api/model/model.dart

```diff
@@ -646,7 +646,8 @@ sealed class Message {
   final String senderFullName;
   final int senderId;
   final String senderRealmStr;
-  final String subject; // TODO call it "topic" internally; also similar others
+  @JsonKey(name: 'subject')
+  final String topic;
```

Comment:
```
api [nfc]: Rename Message.topic, from "subject"
```

I wondered what would happen if the payload already contained a 'topic' key, and I wasn't immediately sure. I could imagine servers adding one as part of an API migration.

We just ignore it, it turns out, which is definitely fine for before that API migration happens. But if we rejected it, we'd be putting up an obstacle to the migration. Do we want to test that we don't reject it?

```dart
    test('does not reject if servers add a new `topic` field', () {
      check(() => Message.fromJson(baseStreamJson()..['topic'] = 'hello'))
        .returnsNormally();
    });
```

(Similarly for the other renames in this PR, I guess.)

Reply:
In general our API types and their `fromJson` constructors always have the property that they don't notice or care if the JSON they receive has additional properties in it that weren't expected. That's a fact about `json_serializable` / `json_annotation` (there's a flag to get the opposite behavior, `JsonSerializable.disallowUnrecognizedKeys`, but it's off by default), and it's one we rely on regularly for routine forward-compatibility with server changes.

So I'm content to leave that as part of the routine boring behavior of the generated deserialization code, which we don't test.